pkg/frontend/v2/frontend.go

```diff
@@ -190,10 +196,11 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 
 	return f.retry.Do(ctx, func() (*httpgrpc.HTTPResponse, error) {
 		freq := &frontendRequest{
-			queryID:      f.lastQueryID.Inc(),
-			request:      req,
-			userID:       userID,
-			statsEnabled: stats.IsEnabled(ctx),
+			queryID:        f.lastQueryID.Inc(),
+			request:        req,
+			userID:         userID,
+			statsEnabled:   stats.IsEnabled(ctx),
+			isHighPriority: util_query.IsHighPriority(requestParams, timestamp, f.limits.HighPriorityQueries(userID)),
```

Comment:
What about excluding non `instant query` or `range query` requests and always mark them normal priority? Doesn't make much sense to still call `IsHighPriority` for them as the function is hardcoded to get `query` param